WordPress/gutenberg
# More block: Match the design in the editor with the front, resolve the undo/redo issue

## What?

With this pull request, my intention was first to only update the **look** of the block in the editor to match the front. But it turned into a small refactor.

When I tested the block, I found that changing it from using an `input` element to use a `PlainText` (v2) would also solve the problem with the undo/redo, described in issue 58875.

Closes  https://github.com/WordPress/gutenberg/issues/34419, https://github.com/WordPress/gutenberg/issues/58875

## Why?

The block was styled to look like the more tag in the classic editor. This made the block recognisable for user who has used the classic editor. But the block style did not match the front, which arguably made it more difficult to see what the end result would be.

## How?

- Updates the default text in the editor to match the default text on the front. The new text is` (more...)`.
- Removes the editor styles, since the styles made the block look different than on the front.
- Replaces the `input` element with a `PlainText` component with an `a` tag, because the `input` element has some default styles that would need to be overwritten.
- Replaces the `onKeyDown` behaviour with `disableLineBreaks`, and uses `__unstablOnSplitAtEnd` to insert a new paragraph block if the user clicks on `Enter` at the very end of the block's text content.

## Testing Instructions




Activate a block theme.

Deactivate Gutenberg.
Create a new post.
Add some content
Add a more block. Adjust the text and then try to use the undo commands.
Add some content after the more block.

Activate Gutenberg with this PR.
Confirm that there are no block validation errors caused by the changes to the more block.
Confirm that the more block has a new design, it should now look like a plain text with a link.
Adjust the text and then try to use the undo commands. Confirm that the undo now works.

Next, test the block in an archive, using a query loop with a post content block.
It must be a post content block, it will not work with an excerpt.
View the archive on the front, locate the post you created, and confirm that:
- The content added before the more block shows
- The more link shows.
- The content after the more block is not shown.
- Clicking the more link brings you to the post, and the browser scrolls to the content **after** the more block. The more block itself should not be visible on the single post view.

## Screenshots or screencast 
Before:
<img width="652" alt="More block in the editor, centered with capitalized letters, dashes on the left and right, and with a white background." src="https://github.com/WordPress/gutenberg/assets/7422055/7a26c759-573c-4dff-bdfb-e703f26f4958">

After:
<img width="228" alt="More block in the editor, which looks like a plain linked text" src="https://github.com/WordPress/gutenberg/assets/7422055/43c5872f-6c0b-4fbc-821e-a36f36451b04">
